src/vaadin-radio-group.html

```diff
@@ -470,13 +502,43 @@
 
         /** @private */
         _labelChanged(label) {
-          if (label) {
-            this.setAttribute('has-label', '');
+          this._setOrToggleAttribute('has-label', label === 0 || !!label);
+        }
+
+        /** @private */
+        _errorMessageChanged(errorMessage) {
+          this._setOrToggleAttribute('has-error-message', errorMessage === 0 || !!errorMessage);
+        }
+
+        /** @private */
+        _helperTextChanged(helperText) {
+          this._setOrToggleAttribute('has-helper', helperText === 0 || !!helperText);
+        }
+
+        /** @private */
+        _setOrToggleAttribute(name, value) {
+          if (!name) {
+            return;
+          }
+
+          if (value) {
+            this.setAttribute(name, (typeof value === 'boolean') ? '' : value);
           } else {
-            this.removeAttribute('has-label');
+            this.removeAttribute(name);
           }
         }
 
+        /** @private */
+        _setOrToggleHasHelperAttribute() {
+          const slottedNodes = this.shadowRoot.querySelector(`[name="helper"]`).assignedNodes();
+          // Only has slotted helper if not a text node
+          // Text nodes are added by the helperText prop and not the helper slot
+          // The filter is added due to shady DOM triggering this slotchange event on helperText prop change
+          this._hasSlottedHelper = slottedNodes.filter(node => node.nodeType !== 3).length > 0;
+
+          this._setOrToggleAttribute('has-helper', this._hasSlottedHelper ? 'slotted' : this.helperText === 0 || !!this.helperText);
```

Comment:
Why do we check for 0 here, is it a special value?

Reply:
Done.